src/box/box.cc

```diff
@@ -1533,6 +1533,18 @@ box_check_replication_anon(void)
 	return anon;
```

Comment:
I wanted to comment a bit on the commit message statements:

> a fiber for each replica and libev timers

Why do you need libev timers for this? They are only needed really when you can't afford a fiber. If you have a fiber, you can do timers easier. Wait sleep for the given timeout and you have the timer. No?

> Creation of a fiber for each anonymous replica seems to be a meaningless overhead

Fibers cost nothing compared to overhead of a relay thread which we create for each replica. I don't think it would be noticeable at all, tbh. How much simpler would the code be with individual fibers? I mean, right now it doesn't look complex really.

Reply:
> Why do you need libev timers for this?

That's how we decided to implement expiration in the design doc. Turned out that it's more complicated that a separate fiber.

> Fibers cost nothing compared to overhead of a relay thread.

Relay works only for connected replicas and expiration fibers are needed for all ones. But yep, the overhead still seems not critical.

> How much simpler would the code be with individual fibers?

As you already said, each fiber would simply do `sleep` + `gc_erase_consumer`. On the other hand, we would need to create such fibers on every replica disconnection - taking that into account, separate fibers seem to be even a bit more complicated. Added a note about it to the commit message.